### ISSUE TITLE
Single prefill failing with head_dim > 64

### ISSUE THREAD

**Issue by demandal25** [ROCm/flashinfer#83]
### Problem Description

Single prefill is failing for `head_dim > 64`, that is, for `128, 192, 256`. Following is an example when you run `python examples/single_prefill_example.py` from the project root directory.

```
Running configuration:
  qo_len=15
  kv_len=127
  num_qo_heads=8
  num_kv_heads=4
  head_dim=128
  causal=False
  kv_layout=NHD
  pos_encoding_mode=NONE
  logits_soft_cap=8.0
  return_lse=True
  FlashInfer output shape: torch.Size([15, 8, 128]), LSE shape: torch.Size([15, 8])
  Reference output shape: torch.Size([15, 8, 128]), LSE shape: torch.Size([15, 8])
  ✗ FAIL: Output or LSE does not match reference implementation
    Max absolute difference for output: 0.5537109375
    Max absolute difference for LSE: 0.5076847076416016
```

**Comment by diptorupd** [comment]
@demandal25  Thank you for reporting the issue and the reproducer. I am looking at it. One of the issues I see is with index offsetting in `compute_qk`. I will post an update once I have a fix.

**Comment by diptorupd** [comment]
Fixed in #86. Note HEAD DIM 192 is not supported for the reason described in #85
